src/Strategy/Python/Poetry/Common.hs

```diff
@@ -0,0 +1,184 @@
+module Strategy.Python.Poetry.Common (
+  getPoetryBuildBackend,
+  toMap,
+  pyProjectDeps,
+  logIgnoredDeps,
+
+  -- * for testing
+  supportedPyProjectDep,
+  supportedPoetryLockDep,
+) where
+
+import Data.Foldable (asum, for_)
+import Data.Map (Map)
+import Data.Map.Strict qualified as Map
+import Data.Maybe (fromMaybe)
+import Data.Text (Text, toLower)
+import DepTypes (
+  DepEnvironment (EnvDevelopment, EnvOther, EnvProduction, EnvTesting),
+  DepType (GitType, PipType, URLType),
+  Dependency (..),
+  VerConstraint (
+    CEq
+  ),
+ )
+import Effect.Logger (Has, Logger (..), Pretty (pretty), logDebug)
+import Strategy.Python.Poetry.PoetryLock (PackageName (..), PoetryLock (..), PoetryLockPackage (..), PoetryLockPackageSource (..))
+import Strategy.Python.Poetry.PyProject (
+  PoetryDependency (..),
+  PyProject (..),
+  PyProjectBuildSystem (..),
+  PyProjectPoetry (..),
+  PyProjectPoetryDetailedVersionDependency (..),
+  PyProjectPoetryGitDependency (..),
+  PyProjectPoetryPathDependency (..),
+  PyProjectPoetryUrlDependency (..),
+  toDependencyVersion,
+ )
+
+-- | Gets build backend of pyproject.
+getPoetryBuildBackend :: PyProject -> Maybe Text
+getPoetryBuildBackend project = buildBackend <$> pyprojectBuildSystem project
+
+-- | Supported pyproject dependencies.
+supportedPyProjectDep :: PoetryDependency -> Bool
+supportedPyProjectDep (PyProjectPoetryPathDependencySpec _) = False
+supportedPyProjectDep _ = True
+
+-- | Logs all ignored dependencies for debugger.
+logIgnoredDeps :: Has Logger sig m => PyProject -> Maybe PoetryLock -> m ()
+logIgnoredDeps pyproject poetryLock = for_ notSupportedDepsMsgs (logDebug . pretty)
+  where
+    notSupportedDepsMsgs :: [Text]
+    notSupportedDepsMsgs = (<> ": ignored in poetry project. Dependency's source is not supported!") <$> notSupportedDeps
```

Comment:
Really? Why do we prefer the less general one? Is there precedent for this elsewhere in the codebase?

Reply:
It's usually a better idea to tell the reader "I'm working with a list", rather than "I'm working with *some* functor, and it doesn't matter which one".  It's the same reason I'd rather see `Just`, instead of `pure`, although `Maybe` implements `Applicative` with `pure = Just` ([source here](https://hackage.haskell.org/package/base-4.15.0.0/docs/src/GHC-Base.html#line-1015))